### DISCUSSION TITLE
Problem with Constitutive law Linear J2 Plasticity

### DISCUSSION THREAD

**Discussion by AriasRH** [KratosMultiphysics/Kratos#10853] [Q&A] [answered]
Hi everybody ,
we have a problem with an structural stress analysis. When we select the constitutive law 'Linear J2 Plasticity', it appears the next error several times:
Thread #5 caught exception: Error: Check failed because rMaterialProperties.Has(EXPONENTIAL_SATURATION_YIELD_STRESS) is not true

in kratos/applications/ConstitutiveLawsApplication/custom_constitutive/small_strains/plasticity/small_strain_j2_plasticity_3d.cpp:630:SmallStrainJ2Plasticity3D::Check
   kratos/applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:1435:BaseSolidElement::Check

we hope someone may help us with a clue about the possible cause, thanks.

**Comment by rubenzorrilla** [comment]
It seems that you need to provide `EXPONENTIAL_SATURATION_YIELD_STRESS` in your `MaterialParemeters.json`. In any case, I ping @AlejandroCornejo as he will probably know what's going on there.

**Comment by AriasRH** [comment]
thank you

**Reply by AlejandroCornejo** [reply]
As @rubenzorrilla mentioned, seems that you should add this missing variable in the StructuralMaterials.json with a certain value.

Honestly this plasticity law was nos implemented by me. I can help you if you use the Generic isotropic plasticity. What do you want to simulate? Maybe I can give you an application example.

Regards,

Alejandro

**Reply by AriasRH** [reply]
Thanks Alejandro, We´re working in the structural analysis of an Aluminum alloy turbine blade. I'm not sure if we could implement another constitutive law for a plastic analysis (maybe: generic small strain isotropic plasticity).

Regards,
Heriberto

**Comment by AlejandroCornejo** [comment]
Try with this one:

~~~json
{
    "properties" : [{
        "model_part_name" : "Structure.Parts_Solid_Solid_Auto1",
        "properties_id"   : 1,
        "Material"        : {
            "constitutive_law" : {
                "name" : "SmallStrainIsotropicPlasticity3DVonMisesVonMises"
            },
            "Variables"        : {
                "DENSITY"                 : 7850.0,
                "YOUNG_MODULUS"           : 206900000000.0,
                "POISSON_RATIO"           : 0.29,
                "YIELD_STRESS"            : 5500000.0,
                "FRACTURE_ENERGY"         : 1.0e8,
                "HARDENING_CURVE"         : 0
            },
            "Tables"           : {}
        }
    }]
}

~~~

**Reply by AriasRH** [reply]
Thanks Alejandro, I'll try it.
I know that you've been working on this for some years. Do you have any reference about Von Mises plasticity, so that I could cite you?

Regards from Morelia
Heriberto Arias

**Reply by AlejandroCornejo** [reply, marked as answer]
Hey! about Von Mises plasticity, a good reference (not mine) could be https://onlinelibrary.wiley.com/doi/book/10.1002/9780470694626 .

An application of the Kratos implementation has been done in:

https://scholar.google.es/citations?view_op=view_citation&hl=es&user=xsUlk_UAAAAJ&authuser=1&citation_for_view=xsUlk_UAAAAJ:zYLM7Y9cAGgC

https://scholar.google.es/citations?view_op=view_citation&hl=es&user=xsUlk_UAAAAJ&authuser=1&citation_for_view=xsUlk_UAAAAJ:hqOjcs7Dif8C

https://scholar.google.es/citations?view_op=view_citation&hl=es&user=xsUlk_UAAAAJ&authuser=1&citation_for_view=xsUlk_UAAAAJ:WF5omc3nYNoC

**Reply by AriasRH** [reply]
Thank you for everything
I'll read them

Regards 
Heriberto
